lib/SIL/Parser/ParseSIL.cpp

```diff
@@ -1690,6 +1694,23 @@ bool SILParser::parseSILDebugInfoExpression(SILDebugInfoExpression &DIExpr) {
           DIExpr.push_back(NewOperand);
           break;
         }
+        case SILDIExprElement::ConstIntKind: {
+          bool IsNegative = false;
+          if (P.Tok.is(tok::oper_prefix) && P.Tok.getRawText() == "-") {
+            P.consumeToken();
+            IsNegative = true;
+          }
+          int64_t Val;
+          // FIXME: Use the correct diagnostic
+          if (parseInteger(Val, diag::sil_invalid_scope_slot))
```

Comment:
Can you create a new diagnostic message for this? (something like "expecting expecting an integer")